tools/upgrademonitor/cmd/root.go

```diff
@@ -0,0 +1,64 @@
+package cmd
+
+import (
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/celestiaorg/celestia-app/tools/upgrademonitor/internal"
+	"github.com/spf13/cobra"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+var rootCmd = &cobra.Command{
+	Use:   "upgrademonitor",
+	Short: "upgrademonitor monitors that status of upgrades on a Celestia network.",
+	RunE: func(cmd *cobra.Command, args []string) error {
+		conn, err := grpc.Dial(grpcEndpoint, grpc.WithTransportCredentials(insecure.NewCredentials()))
+		if err != nil {
+			return fmt.Errorf("could not connect: %v", err)
+		}
+		defer conn.Close()
+
+		ticker := time.NewTicker(time.Duration(pollFrequency) * time.Second)
+		defer ticker.Stop()
+
+		for {
+			select {
+			case <-ticker.C:
+				resp, err := internal.QueryVersionTally(conn, version)
+				if err != nil {
+					return err
+				}
+				fmt.Printf("version: %v, voting: %v, threshold: %v, total: %v\n", version, resp.GetVotingPower(), resp.GetThresholdPower(), resp.GetTotalVotingPower())
+
+				if internal.IsUpgradeable(resp) {
+					fmt.Printf("the network is upgradeable so publishing %v\n", pathToTransaction)
+					resp, err := internal.Publish(conn, pathToTransaction)
+					if err != nil {
+						return err
+					}
+					fmt.Printf("published transaction: %v\n", resp.TxHash)
+					return nil // stop the upgrademonitor
+				}
+			}
+		}
+	},
+}
+
+func Execute() {
+	// Bind the version variable to the --version flag
+	rootCmd.Flags().Uint64Var(&version, "version", defaultVersion, "version to monitor")
```

Comment:
To make this a little more convenient for the user, you could use the grpc endpoint to query the current version of the chain and assume by default that the version to be upgraded to is the next one.

Currently only the next version number is supported i.e. it has to be monotonically increasing

Reply:
Good point, can do!

Reply:
50fcc9a9e230e04817f60588fa5f6738fcd6f0d5